test/05_device-config.ts

```diff
@@ -0,0 +1,58 @@
+import { expect } from './test-lib/chai';
+import * as fixtures from './test-lib/fixtures';
+import { supertest } from './test-lib/supertest';
+import { version } from './test-lib/versions';
+import { sbvrUtils, permissions } from '@balena/pinejs';
+
+const { api } = sbvrUtils;
+
+describe('generate device config', function () {
+	before(async function () {
+		const fx = await fixtures.load('05-device-config');
+		this.loadedFixtures = fx;
+		this.user = fx.users.admin;
+		this.application = fx.applications.app1;
+	});
+
+	after(async function () {
+		await supertest(this.user).delete(`/${version}/api_key`).expect(200);
+		await fixtures.clean(this.loadedFixtures);
+	});
+
+	describe('using /download-config', function () {
+		it('should be able to create a provisioning key with default name and description', async function () {
```

Comment:
How about adding one more for user provided values so that we have a complete scenario?